onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -1849,14 +1849,14 @@ static ORT_STATUS_PTR OrtCreateValueImplSeqHelper(const OrtValue* const* in, siz
                                    "Sequences must have tensors of the same data type. There was at least one tensor in the input that was different.");
     }
 
-    ORT_API_RETURN_IF_ERROR(OrtCreateValueImplSeqHelperTensor(one_tensor, tensors[idx]));
+    Tensor output_tensor;
+    ORT_API_RETURN_IF_ERROR(OrtCreateValueImplSeqHelperTensor(one_tensor, output_tensor));
+    seq_ptr->Add(std::move(output_tensor));
```

Comment:
I was wondering if we could simply copy construct the input OrtValue after running all the checks and add it to the sequence? There is not a need to create another tensor and repackage it again since we now operate in terms of OrtValues?
Am I missing something?

Reply:
The current implementation does a deep copy of the bytes via `PopulateTensorWithData`, but wouldn't constructing the `OrtValue` just increase the refcount on the internal data? This may work, but I'm not familiar enough with that part of the code to know if it would break anything.

Reply:
You are right and this is what I had in mind. Need to check the API docs.

Reply:
Making a copy of OrtValue should work since even if the original OrtValue supplied by the user is released, we'll have 1 ref remaining (due to the copy).

Reply:
We should be Okay, we actually should never copy data. The only cause for concern if that original OrtValues are created on top of the user buffers, but it was always understood that the user must keep them around.